os/src/SubProcess.scala

```diff
@@ -120,7 +126,8 @@ class SubProcess(
    * limit is given, explicitly destroys the subprocess if it has not completed by
    * the time the timeout has occurred
    */
-  def join(timeout: Long = -1): Boolean = {
+  def join(timeout: Long = -1, timeoutGracePeriod: Long = 1000): Boolean = {
+    // FIXME:
     val exitedCleanly = waitFor(timeout)
```

Comment:
The current logic for `join` (undocumented) returns `true` if the process had already terminated by the time the timeout completes. If the process is required to terminate in anyway, this will be `false`.

Would it be better instead to have `exitedCleanly` also mean "exited within the grace period and accepted `SIGTERM` without needing `SIGKILL`"? I would say that also feels like a "clean exit", in that no forcible termination was required.